File: app/src/main/java/com/charlie/sawl/schedule/ScheduleAdapter.java
package com.charlie.sawl.schedule;

import android.support.v4.content.ContextCompat;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import com.charlie.sawl.R;
import java.util.ArrayList;

public class ScheduleAdapter extends RecyclerView.Adapter<ScheduleAdapter.ScheduleViewHolder> {
    private ArrayList<ScheduleInfo> mValues = new ArrayList<>();

    public void addItem(String mSchedule, String mDate) {
        ScheduleInfo addInfo = new ScheduleInfo();

        addInfo.date = mDate;
        addInfo.schedule = mSchedule;
        addInfo.isHoliday = false;
        addInfo.home = false;

        mValues.add(addInfo);
    }

    public void addItem(String mSchedule, String mDate, boolean home, boolean isHoliday) {
        ScheduleInfo addInfo = new ScheduleInfo();

        addInfo.date = mDate;
        addInfo.schedule = mSchedule;
        addInfo.isHoliday = isHoliday;
        addInfo.home = home;

        mValues.add(addInfo);
    }

    @Override
    public ScheduleViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View mView = LayoutInflater.from(parent.getContext()).inflate(R.layout.row_schedule_item, parent, false);

        return new ScheduleViewHolder(mView);
    }

    @Override
    public void onBindViewHolder(final ScheduleViewHolder holder, int position) {
        ScheduleInfo mInfo = getItemData(position);

        holder.mDate.setText(mInfo.date);
        holder.mSchedule.setText(mInfo.schedule);

        if (mInfo.isHoliday && mInfo.home) {
            holder.mDate.setTextColor(ContextCompat.getColor(holder.mDate.getContext(), R.color.refresh_red));
        } else if (mInfo.isHoliday) {
            holder.mDate.setTextColor(ContextCompat.getColor(holder.mDate.getContext(), R.color.refresh_red));
        } else if (mInfo.home) {
            holder.mDate.setTextColor(ContextCompat.getColor(holder.mDate.getContext(), R.color.refresh_blue));
        } else {
            holder.mDate.setTextColor(ContextCompat.getColor(holder.mDate.getContext(), R.color.colorSecondaryText));
        }
    }

    @Override
    public int getItemCount() {
        return mValues.size();
    }

    public ScheduleInfo getItemData(int position) {
        return mValues.get(position);
    }

    public class ScheduleViewHolder extends RecyclerView.ViewHolder {
        public final TextView mDate, mSchedule;

        public ScheduleViewHolder(View mView) {
            super(mView);

            mSchedule = mView.findViewById(R.id.list_item_entry_title);
            mDate = mView.findViewById(R.id.list_item_entry_summary);
        }
    }

    public class ScheduleInfo {
        public String date;
        public String schedule;
        public boolean isHoliday, home;
    }
}

File: app/src/main/kotlin/com/charlie/sawl/schedule/ScheduleAdapter.kt
package com.charlie.sawl.schedule

import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.charlie.sawl.R
import java.util.ArrayList

class ScheduleAdapter : RecyclerView.Adapter<ScheduleAdapter.ScheduleViewHolder>() {
    private val mValues = ArrayList<ScheduleInfo>()

    fun addItem(mSchedule: String, mDate: String) {
        val addInfo = ScheduleInfo()

        addInfo.date = mDate
        addInfo.schedule = mSchedule
        addInfo.isHoliday = false
        addInfo.home = false

        mValues.add(addInfo)
    }

    fun addItem(mSchedule: String, mDate: String, home: Boolean, isHoliday: Boolean) {
        val addInfo = ScheduleInfo()

        addInfo.date = mDate
        addInfo.schedule = mSchedule
        addInfo.isHoliday = isHoliday
        addInfo.home = home

        mValues.add(addInfo)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ScheduleViewHolder {
        val mView = LayoutInflater.from(parent.context).inflate(R.layout.row_schedule_item, parent, false)

        return ScheduleViewHolder(mView)
    }

    override fun onBindViewHolder(holder: ScheduleViewHolder, position: Int) {
        val mInfo = getItemData(position)

        holder.mDate.text = mInfo.date
        holder.mSchedule.text = mInfo.schedule

        if (mInfo.isHoliday && mInfo.home) {
            holder.mDate.setTextColor(ContextCompat.getColor(holder.mDate.context, R.color.refresh_red))
        } else if (mInfo.isHoliday) {
            holder.mDate.setTextColor(ContextCompat.getColor(holder.mDate.context, R.color.refresh_red))
        } else if (mInfo.home) {
            holder.mDate.setTextColor(ContextCompat.getColor(holder.mDate.context, R.color.refresh_blue))
        } else {
            holder.mDate.setTextColor(ContextCompat.getColor(holder.mDate.context, R.color.schedule_normal_text))
        }
    }

    override fun getItemCount(): Int {
        return mValues.size
    }

    fun getItemData(position: Int): ScheduleInfo {
        return mValues[position]
    }

    inner class ScheduleViewHolder(mView: View) : RecyclerView.ViewHolder(mView) {
        val mDate: TextView = mView.findViewById(R.id.list_item_entry_summary)
        val mSchedule: TextView = mView.findViewById(R.id.list_item_entry_title)
    }

    inner class ScheduleInfo {
        var date: String? = null
        var schedule: String? = null
        var isHoliday: Boolean = false
        var home: Boolean = false
    }
}